doc/src/manual/faq.md

```diff
@@ -42,6 +42,9 @@ You can get a complete list of the public symbols from a module with `names(MyMo
 
 Package authors are encouraged to define their public API similarly.
 
+In addition, the documented and semantically observable behaviors of the `julia` CLI driver
```

Comment:
I think it's okay to say that the documented behaviors of the `Julia` driver are public API, since that's consistent with the rest of this FAQ entry (which says that documented behaviors of public symbols are public API). But "semantically observable behaviors" seems too vague and hard to define.

```suggestion
In addition, the documented behaviors of the `julia` CLI driver
```

Reply:
That would have prevented the `--math-mode=fast` removal.

Reply:
I think I'm misunderstanding what you've written here. Which of the following interpretations is correct?

Interpretation 1:
- The documented behaviors of the `julia` driver are public API.
- The semantically observable behaviors of the `julia` driver are public API.

Interpretation 2:
- A behavior of the `julia` driver is public API iff the behavior is documented and the behavior is semantically observable.

When I read the above text, I interpreted it as interpretation 1.

Reply:
Interpretation 2 is what I intended. In particular options that are non-semantic (optimization levels, fastmath, etc.) are not guaranteed to work the same across versions, even if the documentaiton tells you what they do.

Reply:
Let's reword the sentence to get rid of the ambiguity?

Reply:
How about something along the lines of:

> A behavior of the `Julia` CLI driver is only part of the public API if the behavior is documented and the behavior is semantically observable.

Reply:
And it might also be helpful to add the sentence that you wrote above, since I think it gives some good illustrative examples:

> For example, options that are non-semantic (optimization levels, fastmath, etc.) are not guaranteed to work the same across versions, even if the documentaiton tells you what they do.